src/main/java/gregtech/integration/jei/basic/GTOreCategory.java

```diff
@@ -0,0 +1,140 @@
+package gregtech.integration.jei.basic;
+
+import gregtech.api.gui.GuiTextures;
+import gregtech.api.util.GTJEIUtility;
+import gregtech.api.util.GTStringUtils;
+import gregtech.api.util.GTUtility;
+import gregtech.api.worldgen.config.OreDepositDefinition;
+import gregtech.api.worldgen.config.WorldGenRegistry;
+import gregtech.integration.jei.utils.render.ItemStackTextRenderer;
+import mezz.jei.api.IGuiHelper;
+import mezz.jei.api.gui.IDrawable;
+import mezz.jei.api.gui.IGuiItemStackGroup;
+import mezz.jei.api.gui.IRecipeLayout;
+import mezz.jei.api.ingredients.IIngredients;
+import mezz.jei.api.recipe.IRecipeWrapper;
+import net.minecraft.client.Minecraft;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+
+public class GTOreCategory extends BasicRecipeCategory<GTOreInfo, GTOreInfo> {
+
+    protected final IDrawable slot;
+    protected OreDepositDefinition definition;
+    protected String veinName;
+    protected int minHeight;
+    protected int maxHeight;
+    protected int outputCount;
+    protected int weight;
+    protected final int FONT_HEIGHT = Minecraft.getMinecraft().fontRenderer.FONT_HEIGHT;
```

Comment:
You could extract this to base because nearly everyone is going to needed it.

Optional